tools/importer/cmsis_imorter.py

```diff
@@ -0,0 +1,143 @@
+import os, json, stat, sys, shutil, errno, subprocess
+from os.path import dirname, abspath, basename, join
+
+ROOT = abspath(join(dirname(__file__), "../.."))
+CMSIS_REPO = "CMSIS_Repo"
+CMSIS_PATH = abspath(join(dirname(__file__), CMSIS_REPO))
+RTOS_UPDATE_BRANCH = "rtos_update"
+
+def del_file(name):
+    result = []
+    search_path = [join(ROOT, 'rtos'), join(ROOT, 'cmsis')]
+    for path in search_path:
+        for root, dirs, files in os.walk(path):
+            if name in files:
+                result.append(os.path.join(root, name))
+    for file in result:
+        print os.path.relpath(file, ROOT)
+        os.remove(file)
+
+def rmtree(top):
+    for root, dirs, files in os.walk(top, topdown=False):
+        for name in files:
+            filename = os.path.join(root, name)
+            os.chmod(filename, stat.S_IWUSR)
+            os.remove(filename)
+        for name in dirs:
+            os.rmdir(os.path.join(root, name))
+    os.rmdir(top)
+
+def copy_file(file, path):
+    try:
+        shutil.copy(file, path)
+    except IOError as e:        
+        if e.errno != errno.ENOENT:
+            raise
+        ## Create directories
+        os.makedirs(os.path.dirname(path))
+        shutil.copy(file, path)
+    print os.path.relpath(path, ROOT)
+    
+def copy_folder(folder, path):    
+    files = os.listdir(folder)    
+    for file in files:
+        abs_src_file = os.path.join(folder, file)
+        if os.path.isfile(abs_src_file):
+            abs_dst_file = os.path.join(path, file)
+            copy_file(abs_src_file, abs_dst_file)
+    
+def run_cmd(command, exit_on_failure=False):
+    """ Passes a command to the system and returns a True/False result once the 
+        command has been executed, indicating success/failure. Commands are passed
+        as a list of tokens. 
+        E.g. The command 'git remote -v' would be passed in as ['git', 'remote', '-v']
+    """
+    return_code = subprocess.call(command, shell=True)
+    
+    if return_code:
+        print("The command %s failed with return code: %s"
+                        %(' '.join(command), return_code))
+        if exit_on_failure:
+            sys.exit(1)
+    
+    return return_code
+
+def remove_repo(folder):
+    os.chdir(abspath(dirname(__file__)))
+    if os.path.exists(folder):
+        rmtree(folder)
+
+def get_repo(repo, branch, folder):
+    """ Get the Repository files from git, at depth level 1
+        repo - Git repository link
+        branch - repository branch
+        folder - folder at which repo will be cloned
+    """
+    remove_repo(folder)
+    clone_cmd = ['git', 'clone', repo, "-b", branch, "--depth", '1', folder]
+    run_cmd(clone_cmd, exit_on_failure=True)
+  
+if __name__ == "__main__":
```

Comment:
Can we also have option to skip the first part and only do the cherry-picking. I'm assuming flow of run script -> fix conflicts -> re-run script to continue cherry picking.

Reply:
If we want to just cherry-pick than we can keep files/folder section as empty. I will have to fix few things for that (git commit and all will fail). Or do you want some command line option?

Reply:
In first step I want to bring new RTX/CMSIS then the script will do the cherry pick but if it fails due to a conflict I want to be able to fix the conflict and continue, that is skip the first part and just continue with the patches.